pkg/monitortests/kubeapiserver/faultyloadbalancer/monitortest.go

```diff
@@ -0,0 +1,216 @@
+	test.monitor = &faultyLBMonitor{}
+	framework.Logf("monitor[%s]: initialized", MonitorName)
+	return nil
+}
+
+func (test *monitorTest) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	return monitorapi.Intervals{}, nil, nil
+}
+
+func (*monitorTest) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
+	return nil, nil
+}
+
+func (test *monitorTest) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	intervals := test.monitor.Filter(finalIntervals)
+	if len(intervals) == 0 {
+		return nil, nil
+	}
+	framework.Logf("monitor[%s]: found %d interesting intervals ", MonitorName, len(intervals))
+
+	callback := &junitCallback{name: "[sig-apimachinery] new and reused connections to kube-apiserver should be handled gracefully during the graceful termination process"}
+	test.monitor.Check(intervals, callback.Evaluate)
+	return callback.Result(), nil
+}
+
+func (*monitorTest) WriteContentToStorage(ctx context.Context, storageDir, timeSuffix string, finalIntervals monitorapi.Intervals, finalResourceState monitorapi.ResourcesMap) error {
+	return nil
+}
+
+func (*monitorTest) Cleanup(ctx context.Context) error {
+	return nil
+}
+
+type junitCallback struct {
+	name     string
+	testCase *junitapi.JUnitTestCase
+}
+
+// constructs the junit test case by taking into account each overlap of
+// a kube-apiserver shutdown interval and an APIUnreachableFromClient interval
+func (junit *junitCallback) Evaluate(shutdown, unreachable monitorapi.Interval) {
+	if junit.testCase == nil {
+		junit.testCase = &junitapi.JUnitTestCase{
+			Name:          junit.name,
+			SystemOut:     fmt.Sprintf("faulty load balancer detected"),
+			FailureOutput: &junitapi.FailureOutput{},
+		}
+	}
+
+	lbType := unreachable.Condition.Locator.Keys[monitorapi.LocatorAPIUnreachableHostKey]
+	msg := fmt.Sprintf("client observed connection error during kube-apiserver rollout, type: %s\nkube-apiserver: %s\nclient: %s\n", lbType, shutdown.String(), unreachable.String())
+	junit.testCase.FailureOutput.Output = fmt.Sprintf("%s\n%s", junit.testCase.FailureOutput.Output, msg)
+}
+
+func (junit *junitCallback) Result() []*junitapi.JUnitTestCase {
+	testCases := make([]*junitapi.JUnitTestCase, 0)
+	if junit.testCase != nil {
+		// TODO: we will use sippy to find occurrences of faulty load
+		// balancers where it flakes. Once we know it's fully passing then
+		// we can remove the flake test case.
+		flake := &junitapi.JUnitTestCase{
+			Name:      junit.name,
+			SystemOut: fmt.Sprintf("faulty load balancer detected"),
+		}
+		testCases = append(testCases, junit.testCase, flake)
+	}
+	return testCases
```

Comment:
Am I reading this correctly that if this is not going to return anything if the test fully passes? From what I can see, on success junit.testCase is nil, because Evaluate is never called. Then we get here, and return an empty slice. 

You need to return a success case as well otherwise your pass rates will be all out of wack.